src/test/java/uk/gov/ons/census/casesvc/messaging/ReceiptReceiverIT.java

```diff
@@ -155,6 +165,144 @@ public void testReceiptEmitsMessageAndEventIsLoggedForCase()
+
+              expectedActualResponses.add(count + 1);
+              expectedActualResponses.add(numberOfReceiptsAndLinkToSend + count + 1);
+            });
+
+    // THEN
+    Case actualCase =
+        pollDatabaseUntilCorrectActualResponseCount(caseId, expectedResponseCount, 100);
+    assertThat(actualCase.getCeActualResponses())
+        .as("ActualResponses Count")
+        .isEqualTo(expectedResponseCount);
+    assertThat(actualCase.isReceiptReceived()).as("Case Receipted").isTrue();
+
+    checkExpectedResponsesEmitted(
+        expectedActualResponses, rhCaseOutboundQueue, caze.getCaseId().toString());
+  }
+
+  private void checkExpectedResponsesEmitted(
+      List<Integer> expectedActualResponses,
+      BlockingQueue<String> rhCaseOutboundQueue,
+      String caseId)
+      throws IOException {
+
+    List<Integer> actualResponsesList =
+        rabbitQueueHelper.collectAllActualResponseCountsForCaseId(rhCaseOutboundQueue, caseId);
+
+    assertThat(actualResponsesList).hasSameElementsAs(expectedActualResponses);
+  }
+
+  private Case pollDatabaseUntilCorrectActualResponseCount(
+      UUID caseID, int expectedActualResponses, int retryAttempts) throws InterruptedException {
+
+    Case actualCase = null;
+
+    for (int i = 0; i < retryAttempts; i++) {
+      actualCase = caseRepository.findByCaseId(caseID).get();
+
+      if (actualCase.getCeActualResponses() >= expectedActualResponses) {
+        return actualCase;
+      }
+
+      System.out.println("Current Actual Count: " + actualCase.getCeActualResponses());
+
+      Thread.sleep(1000);
+    }
+
+    return actualCase;
+  }
+
+  private Message[] buildLinkReceiptedQidToCaseMsgs(String caseId, int count) {
+
+    Message[] qidLinkingMsgs = new Message[count];
+
+    for (int i = 0; i < count; i++) {
+      UacQidLink receiptedUacQid = new UacQidLink();
+      receiptedUacQid.setId(UUID.randomUUID());
+      receiptedUacQid.setBatchId(UUID.randomUUID());
+      receiptedUacQid.setUac("test uac");
+      receiptedUacQid.setQid("21" + i);
+      receiptedUacQid.setActive(false);
+      UacQidLink createdUacQidLink = uacQidLinkRepository.save(receiptedUacQid);
+
+      String expectedQuestionnaireId = createdUacQidLink.getQid();
+      ResponseManagementEvent managementEvent = getTestResponseManagementQuestionnaireLinkedEvent();
+      managementEvent.getEvent().setTransactionId(UUID.randomUUID());
+      UacDTO uac = new UacDTO();
+      uac.setCaseId(caseId);
+      uac.setQuestionnaireId(expectedQuestionnaireId);
+      managementEvent.getPayload().setUac(uac);
+
+      qidLinkingMsgs[i] =
+          MessageBuilder.withBody(convertObjectToJson(managementEvent).getBytes())
+              .setContentType(MessageProperties.CONTENT_TYPE_JSON)
+              .build();
+    }
+
+    return qidLinkingMsgs;
+  }
+
+  @Test
+  public void testLargeNumberWithReceiptsAndLinkingEventsJustInCase() {
+    fail("Not implemented yet");
```

Comment:
Well... this isn't going to work!

Reply:
It did pass, just checked things, I ran all the tests on Friday etc.  Retried that test, it passed.  Drilled into the 'fail' function, looked a bit weird.  Had imported AspectJ not AssertJ.

Reply:
Test not needed anymore as I combined it with the one above as it made most sense